### ISSUE TITLE
Zamiana odpowiedzi na komentarz a głosy w dół

### ISSUE THREAD

**Issue by awaluk** [CodersCommunity/forum.pasja-informatyki.local#245]
Mamy na forum dostępną opcję zamiany odpowiedzi na komentarz, przydaje się na przykład gdy ktoś się pomyli. Odpowiedź może mieć głosy w górę lub w dół, a komentarz tylko glosy w górę. W momencie przenoszenia odpowiedzi do komentarza głosy w dół nie są czyszczone i później wygląda to tak. Należałoby więc przy zmianie odpowiedzi na komentarz czyścić oddane głosy w dół.
![Zrzut ekranu z 2021-03-31 16-28-06](https://user-images.githubusercontent.com/13801608/113160807-1998bd00-923e-11eb-94a8-0380eb938f78.png)

Z zamianą odpowiedzi na komentarz związany jest także problem #200

**Comment by ScriptyChris** [comment]
A dlaczego właściwie komentarze nie mogą mieć głosów negatywnych?

**Comment by awaluk** [comment]
W teorii mogą, można włączyć taką opcję. U nas nigdy nie była włączona, nie wiem jaki był dokładny powód. Mogę podejrzewać, że głównym mogło być to, że głosy w dół i tak nie zepchną komentarza niżej, bo te są sortowane chronologicznie aby nie zaburzyć sensu dyskusji, ale w teorii to samo można by powiedzieć o sensie głosów w górę.

**Comment by event15** [comment]
A czemu nie czyścić wszystkie głosy? I czemu wyświetlanie jest chronologicznie? Na stackoverflow na przykład są siłą głosów ustawiane - na wykopie podobnie

**Comment by awaluk** [comment]
Skoro i odpowiedź i komentarz mają głosy w górę, to nie widzę potrzeby, aby je czyścić. Usuwać trzeba w dół, bo komentarze tych po prostu nie mają, a więc zachowywanie ich jest błędem. Treść i sens posta pozostają bez zmian, więc myślę że ocena w postaci głosów też pozostaje bez zmian.

Wyświetlanie musi być chronologiczne, aby nie zgubić sensu dyskusji - jak napisałem. Zarówno na SO jak i wykopie ustawiane są chronologicznie.

**Comment by ScriptyChris** [comment]
Rozpatrując dwa przypadki odpowiedzi zamienianej na komentarz:
- 1 głos w górę i 3 w dół: wyświetli ocenę -2 obok odpowiedzi. Zamiana na komentarz pokaże 1 głos w górę.
- 3 głosy w górę i 1 w dół: wyświetli ocenę 2 obok odpowiedzi. Zamiana na komentarz pokaże 3 głosy w górę.

Jeśli się nie pomyliłem, to wg mnie przez usunięcie głosów w dół zmieni się prezentowana ocena posta, co może negatywnie wpływać na jego odbiór. Np. odpowiedź z sumarycznie większą liczbą głosów negatywnych będzie prezentowana jako pozytywna, będąc już komentarzem.

**Comment by awaluk** [comment]
Jest w tym trochę prawdy, tyle że gdyby ta odpowiedź od początku była komentarzem, to też nie można byłoby oddać na niego głosów w dół, a tylko w górę. Więc prawdopodobnie efekt byłby taki sam. Chyba, że ktoś oddaje głosy w górę tylko z litości tzn. widzi że ktoś ma w danym poście głosy w dół, więc żeby mu nie było smutno to da też w górę, bez merytorycznej oceny wypowiedzi.

**Comment by ScriptyChris** [comment]
Moim zdaniem w takim przypadku fix powinien polegać na dodaniu możliwości oddawania głosów w dół na komentarz. Ocena komentarza nie musi koniecznie wpływać na jego chronologię w dyskusji, ale już prezentowana ocena powinna być spójna i jeśli odpowiedzi mogą mieć głosy w dół, to i komentarze powinny działać tak samo - a wtedy nie będzie problemu, że przy zamianie odpowiedzi na komentarz znikną głosy, bo po prostu zostaną przeniesione 1:1.

**Comment by awaluk** [comment]
Można o tym pomyśleć skoro tak mówisz. Ewentualnie zrobić jakąś ankietę i zapytać. Tylko to już bardziej dodanie nowego ficzera, niż poprawka buga ;) Choć to prawda, że bug się przy okazji rozwiąże sam. Jeśli jednak głosy w dół dla komentarzy mają być dodane tylko dlatego, żeby się dobrze przeniosło przy zmianie z odpowiedzi, a nie widzimy realnego faktycznego sensu istnienia głosów w dół do komentarzy, to jak dla mnie overkill.

Z założenia jak mówiłem głosy mają odpowiedzi, bo one są sortowane po głosach - dzięki głosom w dół można łatwo zepchnąć na dół te kiepskie. To odpowiedzi powinny być głównie oceniane, a komentarze są dodatkiem, najczęściej tylko dyskusją na temat odpowiedzi. O ile mi się wydaje SO też ma głosy w górę i w dół do odpowiedzi oraz w górę do komentarzy. Nie przypominam sobie, abym widział kiedyś przy komentarzach głos w dół.

**Comment by ScriptyChris** [comment]
> Jeśli jednak głosy w dół dla komentarzy mają być dodane tylko dlatego, żeby się dobrze przeniosło przy zmianie z odpowiedzi, a nie widzimy realnego faktycznego sensu istnienia głosów w dół do komentarzy, to jak dla mnie overkill.

Nie. Jak dla mnie dodanie możliwości głosowania w dół na komentarze będzie uspójnieniem ich względem odpowiedzi. Choć do pełnego uspójnienia trzeba by pewnie podpiąć pod to powiadomienia i przeliczanie głosów na punkty profilowe. Natomiast samoistne rozwiązanie problemu przenosin głosów przy zmianie odpowiedzi na komentarz będzie wtedy przy okazji.

Osobiście nie rozumiem, dlaczego komentarze (czy to u nas, czy na SO) mają mieć tylko głosy w górę, ale już nie w dół. Dla mnie możliwość oceny powinna być w obie strony. Natomiast wpływ tej oceny na pozycję wypowiedzi w dyskusji jest wg mnie osobną sprawą, być może też drugorzędną, choć pewnie przydatną w długich dyskusjach. Rozdzieliłbym więc rozmyślanie nad dodaniem możliwości głosowania w dół na komentarze od rozmyślań nad umożliwieniem wpływania tych głosów na chronologiczną pozycję komentarzy w dyskusji.
